DataProcessing/datax-host/src/main/scala/datax/processor/CommonProcessorFactory.scala

```diff
@@ -479,39 +477,120 @@ object CommonProcessorFactory {
+            val latencyMetrics = Map(s"Latency-Blobs" -> latencyInSeconds)
+            postMetrics(latencyMetrics)
+            latencyMetrics++processedMetrics
+          }
+          else{
+            processedMetrics
+          }
+        }
 
+        def processPartition(v: (String, HashSet[FileInternal])) = {
+          val par = v._1
+          val paths = v._2.toArray
+          processBlobs(paths, outputPartitionTime, par+namespace, par)
+        }
+
+       batchLog.warn(s"Start batch ${batchTime}, output partition time:${outputPartitionTime}, namespace:${namespace}")
+       val t1 = System.nanoTime
+
+//        // Wrap files to FileInternal object
+//       val internalFiles =  pathsRDD.map(file => BatchBlobInput.filePathToInternalFileInfo(file, dict))  // get Partition and InputTime in properties if fileTimeRegex is configured
+
+//        val paths = internalFiles.map(_.inputPath).collect()
+//        postMetrics(Map(s"InputBlobs" -> paths.size.toDouble))
+//        batchLog.warn(s"InputBlob count=${paths.size}");
+
+        val pathsGroups = BlobPointerInput.pathsToGroups(rdd = pathsRDD,
+          jobName = dict.getAppName(),
+          dict = dict,
+          outputTimestamp = outputPartitionTime)
+        val pathsFilteredGroups = BlobPointerInput.filterPathGroups(pathsGroups)
+        val pathsCount = pathsFilteredGroups.aggregate(0)(_ + _._2.size, _ + _)
+        //try {
+        val result =
+          if (pathsCount > 0) {
+            batchLog.warn(s"Loading filtered blob files count=$pathsCount, First File=${pathsFilteredGroups.head._2.head}")
+            if (pathsFilteredGroups.length > 1)
+              Await.result(FutureUtil.failFast(pathsFilteredGroups
+                .map(kv => Future {
+                  processPartition(kv)
+                })), 5 minutes).reduce(DataMerger.mergeMapOfDoubles)
+            else
+              processPartition(pathsFilteredGroups(0))
+          }
+          else {
+            batchLog.warn(s"No valid paths were found to process for this batch")
+            val bShouldForceNonEmptyBatch = ConfigManager.getActiveDictionary().getOrElse(JobArgument.ConfName_ForceNonEmptyBatch, "false").toBoolean
+            if(bShouldForceNonEmptyBatch)
+            {
+              throw new Exception("No valid paths found to process for this batch and empty batches are not allowed")
+            }
+            Map[String, Double]()
+          }
         val batchProcessingTime = (System.nanoTime - t1) / 1E9
 
         val metrics = Map[String, Double](
           "BatchProcessedET" -> batchProcessingTime
         )
-        processedMetrics ++ metrics
+
+        postMetrics(metrics)
+        batchLog.warn(s"End batch ${batchTime}, output partition time:${outputPartitionTime}, namespace:${namespace}")
+
+        metrics ++ result
+
+//        val blobStorageKey = ExecutorHelper.createBlobStorageKeyBroadcastVariable(paths.head, spark)
```

Comment:
These changes will break the streaming app. As discussed offline, please refactor and if required add a flag to distinguish between a streaming job and batch job